contracts/utils/types/Time.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {Math} from "../math/Math.sol";
+import {SafeCast} from "../math/SafeCast.sol";
+
+/**
+ * @dev This library provides helpers for manipulating time-related objects.
+ *
+ * It uses the following types:
+ * - `uint48` for timepoints
+ * - `uint32` for durations
+ *
+ * While the library doesn't provide specific types for timepoints and duration, it does provide:
+ * - a `Delay` type to represent duration that can be programmed to change value automatically at a given point
+ * - additional helper functions
+ */
+library Time {
+    using Time for *;
+
+    /**
+     * @dev Get the block timestamp as a Timepoint
+     */
+    function timestamp() internal view returns (uint48) {
+        return SafeCast.toUint48(block.timestamp);
+    }
+
+    /**
+     * @dev Get the block number as a Timepoint
+     */
+    function blockNumber() internal view returns (uint48) {
+        return SafeCast.toUint48(block.number);
+    }
+
+    /**
+     * @dev Check if a timepoint is set, and in the past
+     */
+    function isSetAndPast(uint48 timepoint, uint48 ref) internal pure returns (bool) {
+        return timepoint != 0 && timepoint <= ref;
+    }
+
```

Comment:
I've seen multiple comparisons with `0` and perhaps this abstraction makes sense:

```solidity
function isSet(uint48 timepoint) internal pure returns (bool) {
  return timepoint != 0;
}
```

Reply:
We used to have it. It was removed during the library simplification. I'm ok to re-add it.

Reply:
I'd be in favor of re-adding it. I think the comparison with `0` is arbitrary and an extension may want to add extra states (eg. `_DONE_TIMESTAMP` being 0 in Timelock), so an overrideable place for it would be great.

Reply:
FWIW, `_DONE_TIMESTAMP` was a bad idea in TimelockController.